### PR TITLE
[iOS/tvOS] Fix target OS case-sensitivity

### LINKED ISSUE #50379: TargetOS is case-sensitive for iOSSimulator when running tests
`./dotnet.sh build src/libraries/System.Threading.Timer/tests/ /t:test /p:targetos=iossimulator /p:targetarchitecture=x64` will fail, but `./dotnet.sh build src/libraries/System.Threading.Timer/tests/ /t:test /p:targetos=iOSSimulator /p:targetarchitecture=x64` will succeed. MSBuild otherwise isn't case-sensitive, so this should probably just be fixed.

### PR DESCRIPTION
**Pull request by MaximLipnin**
Fixes https://github.com/dotnet/runtime/issues/50379

### CODE REVIEW

File: eng/testing/AppleRunnerTemplate.sh [hunk 1]
```diff
@@ -11,29 +11,35 @@ XHARNESS_CMD="test"
 XHARNESS_OUT="$EXECUTION_DIR/xharness-output"
 XCODE_PATH=$(xcode-select -p)/../..
 
+MACCATALYST="maccatalyst"
+IOS_SIMULATOR="iossimulator"
+IOS="ios"
+TVOS_SIMULATOR="tvossimulator"
+TVOS="tvos"
+
```
Comment (lambdageek) [inline eng/testing/AppleRunnerTemplate.sh:19]: What do you think about

```suggestion
TARGET_OS=`echo "$TARGET_OS" | tr '[:upper:]' '[:lower:]'`
```

Comment (MaximLipnin) [reply eng/testing/AppleRunnerTemplate.sh:19]: Thanks Aleksey, even though `TARGET_OS` value should already be in lower case here, your suggestion to fix it in the template looks more general.

**Review by akoeplinger — CHANGES_REQUESTED** [review]

File: eng/testing/AppleRunnerTemplate.sh [hunk 2]
```diff
@@ -11,29 +11,35 @@ XHARNESS_CMD="test"
 XHARNESS_OUT="$EXECUTION_DIR/xharness-output"
 XCODE_PATH=$(xcode-select -p)/../..
 
+MACCATALYST="maccatalyst"
+IOS_SIMULATOR="iossimulator"
+IOS="ios"
+TVOS_SIMULATOR="tvossimulator"
+TVOS="tvos"
+TARGET_OS=`echo "$TARGET_OS" | tr '[:upper:]' '[:lower:]'`
```
[the remark this answers is not in the document]
Comment (MaximLipnin) [reply eng/testing/AppleRunnerTemplate.sh:19]: Heh, It was sorta previous fix :) Theoretically we can call the template from somewhere else and we will have not to forget to make target os lowercase there as well.

[the remark this answers is not in the document]
Comment (akoeplinger) [reply eng/testing/AppleRunnerTemplate.sh]: True, but I think that's not very likely :)

[the remark this answers is not in the document]
Comment (MaximLipnin) [reply eng/testing/AppleRunnerTemplate.sh]: Updated, thanks

**Review by akoeplinger — APPROVED** [review]

**Pull request merged**